tests/thread_msg_block_race/tests/01-run.py

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2019 Freie Universität Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import sys
+from testrunner import run
+from pexpect import TIMEOUT
+
+
+def testfunc(child):
+    res = child.expect([TIMEOUT, "FAILED ASSERTION."])
+    # we actually want the timeout here. The application runs into an assertion
+    # pretty quickly when failing and runs forever on success
+    assert(res == 0)
```

Comment:
Don't put parenthesis with `assert`, use `assert res == 0` !

Reply:
This looks ugly and isn't how I write assertions in C code.

Reply:
In this case I disagree with Python style. We also don't use `sizeof` without parans in C even though you can.

Reply:
(also not keeping to that style saved my ass when `print` suddenly became a function when they switched to Python 3 ;-P)

Reply:
There is nothing about that in [PEP 8](https://www.python.org/dev/peps/pep-0008/) btw...

Reply:
Maybe but look at the [Python documentation](https://docs.python.org/3/reference/simple_stmts.html), `assert` is a statement, like `raise` or `return`. Do you use them with parenthesis ?
```python
raise(Exception('this is an exception'))

def example():
    return(42)
```

To quote [the doc section about assert](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement):
```
The simple form, assert expression, is equivalent to

if __debug__:
    if not expression: raise AssertionError

The extended form, assert expression1, expression2, is equivalent to

if __debug__:
    if not expression1: raise AssertionError(expression2)


```

Reply:
Yepp, but if and return are not function-like. As I said: we don't follow the common style guidelines for `sizeof` either and I'm not gonna start. Can we please stop this discussions about minor personal style disagreements and focus on the actual quality of the code. The maintainer guidelines even point out to look at this first and at style later. Once https://github.com/RIOT-OS/RIOT/pull/10431 is merged I would replace this with an assertRaises() anyway so this discussion isn't even future-proof.

Reply:
> but if and return are not function-like

In Python, `assert` is not considered as a function. That's my point. Which is not about style, but about correct the use of a programming language. Also I don't understand why you talk about `sizeof`. C is a different language, with different rules.

> I would replace this with an assertRaises()

I guess you meant `assertEqual`.

Reply:
I also agree that `assert` is normally without parenthesis in python. But I will not block the test for this.